sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveClientSuite.scala

```diff
@@ -87,6 +90,18 @@ class HiveClientSuite(version: String)
     assert(filteredPartitions.size == testPartitionCount)
   }
 
+  test(s"getPartitionsByFilter should throw an exception if $partPruningFallbackKey=false") {
+    withSQLConf(SQLConf.HIVE_METASTORE_PARTITION_PRUNING_FALLBACK.key -> "false") {
```

Comment:
Also need to set `HIVE_METASTORE_PARTITION_PRUNING.key` to `true`

Reply:
Ok will do.

Reply:
Done.